Fix a bug when "flutter build apk --build-number=2 --build-name=1.0.2" doesn't work as expected.

kangwang1988 opened the pull request:
Currently, when using flutter build apk --build-number=2 --build-name=1.0.2 command to build apk with specific build number and names, the local.properties set properly could be reset because of 
call stack as below:
First set (works as expected, the build info is used and set)
```
 updateLocalProperties(project: project, buildInfo: buildInfo);
buildGradleProject
```

Second set(works unexpectedly for FlutterPluginVersion.managed and FlutterPluginVersion.v2)
```
updateLocalProperties
updateLocalProperties(project: flutterProject);
_gradleProject
_buildGradleProjectV2
buildGradleProject
```

When the second set is triggered, as the build info is missing, the build-number and name will be useless and the local.properties will be reset.

PS.
Though I want to write a test to avoid the later regression, I found it hard for me to do it in a concise way now.

kangwang1988 committed:
Fix a bug when "flutter build apk --build-number=2 --build-name=1.0.2" doesn't work as expected.

kangwang1988 committed:
Fix a bug when "flutter build apk --build-number=2 --build-name=1.0.2" doesn't work as expected.

zoechi commented:
Is this supposed to fix #22788?

kangwang1988 commented:
@zoechi 
I guess so. The current behavior now seems to be a bug. The local.properties set priorly is reset later, which makes the user passed arguments not working.

zoechi commented:
Is this overlapping with #23826 ?

Hixie committed:
Improve StreamBuilder documentation (#23713)

engine-flutter-autoroll committed:
25d99cd25 Use software renderer in embedder unittests (flutter/engine#6928) (#24749)


https://github.com/flutter/engine/compare/99e73d8c64ed...25d99cd25780

git log 99e73d8c64ed..25d99cd25780 --no-merges --oneline
25d99cd25 Use software renderer in embedder unittests (flutter/engine#6928)


The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/&#43;/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff, who should
be CC&#39;d on the roll, and stop the roller if necessary.

engine-flutter-autoroll committed:
eebc6a589 Roll Skia to 988f54e4689b3d9fb6d5220e02b418f5cfa2b6db (flutter/engine#6931) (#24751)


https://github.com/flutter/engine/compare/25d99cd25780...eebc6a589585

git log 25d99cd25780..eebc6a589585 --no-merges --oneline
eebc6a589 Roll Skia to 988f54e4689b3d9fb6d5220e02b418f5cfa2b6db (flutter/engine#6931)


The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/&#43;/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff, who should
be CC&#39;d on the roll, and stop the roller if necessary.

Hixie committed:
Fix typo in URL (#24631)

chinmaygarde committed:
Roll engine to 84e3b137a16d3c673e89b340fbd2b097a4b5a3ba (#24759)

Picks up changes that unblock the Fuchsia roll https://github.com/flutter/engine/pull/6949.

jason-simmons commented on packages/flutter_tools/lib/src/android/gradle.dart:
```diff
@@ -83,17 +83,17 @@ Future<File> getGradleAppOut(AndroidProject androidProject) async {
   final FlutterProject flutterProject = await FlutterProject.current();
   final String gradle = await _ensureGradle(flutterProject);
-  updateLocalProperties(project: flutterProject);
+  updateLocalProperties(project: flutterProject,buildInfo: buildInfo);
```
style nit: add a space after the comma

jason-simmons commented on packages/flutter_tools/lib/src/android/gradle.dart:
```diff
@@ -327,7 +327,7 @@ Future<void> _buildGradleProjectV2(
     BuildInfo buildInfo,
     String target) async {
-  final GradleProject project = await _gradleProject();
+  final GradleProject project = await _gradleProject(buildInfo:buildInfo);
```
style nit: add a space after the colon

Hixie committed:
Some more documentation around WidgetBuilder and close friends. (#24731)

Hixie committed:
Improve CircleAvatar documentation (#24729)

Hixie committed:
Update dartdocs devicelab test to handle current analyzer output (#24582)

engine-flutter-autoroll committed:
5bf4deb43 Update Dartdoc for SceneBuilder#addPlatformView (flutter/engine#6957) (#24794)


https://github.com/flutter/engine/compare/84e3b137a16d...5bf4deb4353c

git log 84e3b137a16d..5bf4deb4353c --no-merges --oneline
5bf4deb43 Update Dartdoc for SceneBuilder#addPlatformView (flutter/engine#6957)


The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/&#43;/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff, who should
be CC&#39;d on the roll, and stop the roller if necessary.

Hixie committed:
Handle the case where there are no missing dartdocs too... (#24801)

engine-flutter-autoroll committed:
2ab019412 Use application_kernel_asset in flutter_tester (flutter/engine#6972) (#24803)


https://github.com/flutter/engine/compare/5bf4deb4353c...2ab019412dce

git log 5bf4deb4353c..2ab019412dce --no-merges --oneline
2ab019412 Use application_kernel_asset in flutter_tester (flutter/engine#6972)


The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/&#43;/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff, who should
be CC&#39;d on the roll, and stop the roller if necessary.

engine-flutter-autoroll committed:
Roll engine 2ab019412dce..be973ea19612 (2 commits) (#24810)

* 2f706bea0 Fix EXIF orentation problem (flutter/engine#6974)

* be973ea19 Roll Skia to bb37aba39910ef1a95cdcbb5b85d99cdabf3c362 (flutter/engine#6975)

tvolkert committed:
Add link to master docs staging site (#24815)

gspencergoog committed:
Remove snippets snapshotting hack from dartdoc generation. (#24812)

Now that dartdoc automatically generates snapshots for external dart tools, I can remove my path hack from the dartdoc_options.yaml file.

This will allow other packages to again build dartdocs (e.g. plugins) that link to Flutter's dartdocs, and allow me to re-enable dartdoc's cross-linking test that was disabled because of this hack.

kangwang1988 committed:
Flutter code style fix.

kangwang1988 committed:
Flutter code style fix.

gspencergoog committed:
Merge analaytics from docs site and flutter.io site. (#24825)

GaryQian committed:
Roll engine be973ea1961..72c7a756722 (#24828)

72c7a7567 Prepend [NSLocale currentLocale] for first locale on iOS to ensure countryCode exists. Allow language-only locales.

Commit:
Updating readme templates for newly created projects (#24725)

We hypothesize that many first-time users creating a new project will look at the README.md as a starting point for "what next". As we've updated our documentation particularly for new users, having better links to our codelabs and cookbooks will help them be productive.

tvolkert committed:
Remove superfluous "install" (#24836)

dnfield committed:
Smoke test building IPA and APK on supported platforms (#24601)

* build tests - AOT on all, APK on Linux, IPA on Mac

engine-flutter-autoroll committed:
7375a0f41 Roll Skia to 57d29eaf2ed7518983d9e91fd5219f4cfc181f88 (flutter/engine#7024) (#24874)


https://github.com/flutter/engine/compare/72c7a7567228...7375a0f414bd

git log 72c7a7567228..7375a0f414bd --no-merges --oneline
7375a0f41 Roll Skia to 57d29eaf2ed7518983d9e91fd5219f4cfc181f88 (flutter/engine#7024)


The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/&#43;/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Hixie committed:
Clarify dart:ui dependencies in foundation library (#24868)

Commit:
Add a flutter-attach entry point for fuchsia (#24878)

GaryQian committed:
Fix typo in documentation (#25003)

goderbauer committed:
Add animations to SliverAppBar doc (#25091)

the4thfloor commented:
To make it more stable, we should add `@required` for the `BuildInfo buildInfo` property of the `updateLocalProperties` method (line 214)

goderbauer committed:
Fix semantics compiler for offstage children (#24862)

Fixes #20313.

jslavitz committed:
Adds small fix to Cupertino navigation demo

liyuqian committed:
Change network image URL in doc (#25096)

This fixes https://github.com/flutter/flutter/issues/25092

chalin committed:
Replace deprecated link in docs (#25120)

Followup to https://github.com/flutter/website/pull/1949, as requested by @Hixie in https://github.com/flutter/website/issues/1634.

cc @kwalrath

Commit:
Update Switch doc: disabled state (#24941)

tvolkert committed:
Ensure that cache dirs and files have appropriate permissions (#24669)

Fixes https://github.com/flutter/flutter/issues/24413

stereotype441 committed:
Remove deprecated lint "prefer_bool_in_asserts". (#24890)

In Dart 2, asserts no longer accept non-bool values so this rule is
made redundant by the Dart analyzer's basic checks and is no longer
necessary.

The rule will be removed in a future Linter release.

tvolkert committed:
Include error message in crash reports (#24632)

This modifies our flutter_tools crash reports to include the error
message. This error message may contain personally identifying
information (PII), such as a file system path on the developer's
local machine that may contain user names, project code names,
etc. To disable crash reporting, the developer can run the
following command:

    flutter config --no-analytics

For more information on what gets reported during crashes of the
flutter tool, see:

https://github.com/flutter/flutter/wiki/Flutter-CLI-crash-reporting

Commit:
ensure lastBuildTimestamp is set before early return (#25058)

xster committed:
Temporarily add the incorrectly filtered exoplayer gradle dependency (#25186)

goderbauer committed:
Remove offstage wording from KeepAlive (#24881)

What KeepAlive does is different from the Offstage widget. Let's not confuse these two and be more speicific in the wording for KeepAlive.

goderbauer committed:
Fix debugPrint(null) to not crash (#24942)

goderbauer committed:
Don't crash if pinned & floating AppBar has less than minExtent remainingPaintExtent (#25048)

Fixes https://github.com/flutter/flutter/issues/21887

DanTup committed:
Don't require the AVD folder to exist in order to run `flutter emulators` (#25154)

Fixes #24750.

Commit:
Add a check for build methods that return context.widget (#25046)

Commit:
Fuchsia multiple devices and target (#24953)

gspencergoog committed:
Adds favicon to Dash/Zeal docset, adds OpenSearch metadata. (#25178)

This sets the favicon for the offline Dash/Zeal docs.

Also, sets up the OpenSearch Description metadata file so that people can create custom search shortcuts for the API docs site.

Fixes #6412

Commit:
Fixed an InheritedWidget doc typo (#25168)

justinmc committed:
TextFormField cursor params (#24635)

* cursor fields on textformfield

* Test pass-through of cursor properties

jimmycasey committed:
Fixed Spelling. (#25217)

Commit:
Handle a TabBarView special case: last tab deleted before animation ends (#24892)

Commit:
Revert "Ensure that cache dirs and files have appropriate permissions" (#25240)

engine-flutter-autoroll committed:
Roll engine 7375a0f414bd..25b58c0c67dc (198 commits) (#25236)

goderbauer committed:
Do not fade out text for pinned & floating AppBar (#25051)

xster committed:
Add navigatorKey to CupertinoTabView (#25183)

trevorwang committed:
Support TextField multi-line hint text #20941 (#24976)

engine-flutter-autoroll committed:
Roll engine 25b58c0c67dc..abb1d3a6c0b2 (3 commits) (#25258)

* 31e6861d4  Reland "Compile libcxx and libcxxabi for Android (#6886)" (flutter/engine#7160)

* 648b844b7 Roll buildroot to 4cb5a74c9612b71b917997f46e97da6d1051eab4 (flutter/engine#7194)

* abb1d3a6c Roll buildroot to 8e538639660413490ea9261eee84864005e240f4 (flutter/engine#7195)

DanTup committed:
Add some a basic debug stepping tests (#24515)

DaveShuckerow committed:
Add ipv6 and observatory port support to the attach command. (#24537)

DanTup committed:
Make doctor output consistent between VS Code/IntelliJ/Android Studio when plugins are missing (#25269)

* Update VS Code validator to match Android Studio

- Now shows a tick (instead of partial) if installed
- Now shows a cross (instead of dot) if extension is not installed

Fixes #22931.

jcollins-g committed:
Allow snippets tool to be run from arbitrary CWDs (#25243)

* Allow snippets tool to be run from arbitrary CWDs

* Drop use of absolute when using Platform.executable (path is not relative, might require PATH resolution)

* canonicalize paths

* Use path.fromUri consistently to resolve file URIs

* Force commit to try to kick Cirrus

* Force commit to kick Cirrus, again

justinmc committed:
Validate style in TextField (#24587)

* Validate style in TextField

* Fix analyze problems

* Use assert and move to top

* Simplified assertion

engine-flutter-autoroll committed:
e5195ee47 Remove unnecessary includes of Ganesh headers (flutter/engine#7189) (#25282)


https://github.com/flutter/engine/compare/abb1d3a6c0b2...e5195ee479c4

git log abb1d3a6c0b2..e5195ee479c4 --no-merges --oneline
e5195ee47 Remove unnecessary includes of Ganesh headers (flutter/engine#7189)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Commit:
Run flutter tests through mini test engine when run directly (flutter run -t test_file) (#24930)

DaveShuckerow committed:
Revert "Add ipv6 and observatory port support to the attach command." (#25288)

* Revert "e5195ee47 Remove unnecessary includes of Ganesh headers (flutter/engine#7189) (#25282)"

This reverts commit f198d66332f0f95f1932d35457801809b6e1d543.

* Revert "Validate style in TextField (#24587)"

This reverts commit 9a8e2f0c4b07d81fb3a0d3274443d26a7f34e9fe.

* Revert "Allow snippets tool to be run from arbitrary CWDs (#25243)"

This reverts commit 4a110b6227a10637e1671f1f7ced7faf4da45d5e.

* Revert "Make doctor output consistent between VS Code/IntelliJ/Android Studio when plugins are missing (#25269)"

This reverts commit e29b023a6bbbd6b106aaf8802db3192ff3d3cf9c.

* Revert "Add ipv6 and observatory port support to the attach command. (#24537)"

This reverts commit 9150b3f031a9b1f743e25b9516a72a0bb3064adc.

goderbauer committed:
Remove uses-material-design from hello_world example (#25300)

goderbauer committed:
Call mark* methods before attaching child (#25239)

engine-flutter-autoroll committed:
Roll engine e5195ee479c4..571f47457c0e (4 commits) (#25302)

* 2bd75ae8b Support querying display refresh rate in engine (flutter/engine#7002)

* 793a08bbf Add .woff file to binary format (flutter/engine#7202)

* daf0ededb Roll src/third_party/skia 6417060b1c23..340aeda6b3ba (41 commits) (flutter/engine#7205)

* 571f47457 Update marked text before selection so textInRange is never called with (flutter/engine#6989)

johnsonmh committed:
[Material] Theme-able elevation on dialogs. (#24169)

* Themable elevation on dialogs.

* Added `BackgroundColor` in widget + theme

* Addressing Comments

* Fix test name

* Add debugFillProperties test

justinmc committed:
Text field style merge (#24449)

* Merge TextField style with global style so that style changes don't override unspecified fields

* Test that a style param is merged with the theme

* Test a few more style properties

* Analysis fix

engine-flutter-autoroll committed:
Roll engine 571f47457c0e..921b1ccebb8f (4 commits) (#25314)


https://github.com/flutter/engine/compare/571f47457c0e...921b1ccebb8f

git log 571f47457c0e..921b1ccebb8f --no-merges --oneline
921b1cceb Roll src/third_party/skia 58fc5d19ad1b..a69b10312977 (3 commits) (flutter/engine#7210)
37bf807a5 Roll src/third_party/skia 17c574a193ad..58fc5d19ad1b (1 commits) (flutter/engine#7209)
0ea994a64 Minor Docs to runtime controller WindowData (flutter/engine#6991)
41dc628d6 Roll src/third_party/skia 340aeda6b3ba..17c574a193ad (8 commits) (flutter/engine#7206)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

justinmc committed:
obscureText and enableInteractiveSelection defaults (#24527)

* obscureText true defaults to disabling selection

* Tests and comments for selectable text field

* Improve selection docs

* Refactor so that all enableInteractiveSelection params are null by default, delegate to selectionEnabled

* Fix selection param macros

justinmc committed:
Revert "obscureText and enableInteractiveSelection defaults (#24527)" (#25335)

This reverts commit f8f5953d8002cb17b861486dff93e5456ca54efd.

engine-flutter-autoroll committed:
a549981da Roll src/third_party/skia a69b10312977..5eb29448dfbd (1 commits) (flutter/engine#7211) (#25333)


https://github.com/flutter/engine/compare/921b1ccebb8f...a549981da706

git log 921b1ccebb8f..a549981da706 --no-merges --oneline
a549981da Roll src/third_party/skia a69b10312977..5eb29448dfbd (1 commits) (flutter/engine#7211)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

JaspervanRiet committed:
Allow detection of taps on TabBar (#23919)

DaveShuckerow committed:
Use stderr instead of stdout to contain errors in flutter attach test (#25305)

gspencergoog committed:
Add imports section to sample code templates, and more docs. (#25184)

Adds documentation for the stateless widget template, and adds an "imports" section to both templates.

jslavitz committed:
Adds support for floating cursor (#24761)

* Adds support for floating cursor!

Commit:
Revert "Adds support for floating cursor (#24761)" (#25352)

This reverts commit 46878d848566769e51f21407de2fead528c1d2c4.

engine-flutter-autoroll committed:
Roll engine a549981da706..101b27da7a1b (4 commits) (#25341)


https://github.com/flutter/engine/compare/a549981da706...101b27da7a1b

git log a549981da706..101b27da7a1b --no-merges --oneline
101b27da7 Roll src/third_party/skia f9e824b8e749..cb6b4bde0c2e (7 commits) (flutter/engine#7215)
eff5e6735 Fallback font match caching to fix emoji lag. (flutter/engine#7208)
18a4e33c2 Downloading and installation of dynamic updates on Android (flutter/engine#7207)
8e56b549a Roll src/third_party/skia 5eb29448dfbd..f9e824b8e749 (10 commits) (flutter/engine#7212)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Commit:
Revert "Use stderr instead of stdout to contain errors in flutter attach test" (#25380)

goderbauer committed:
Revert "Call mark* methods before attaching child (#25239)" (#25382)

This reverts commit 5c1f1d7618e27a3ed5a97971edc95eff9d0ba967.

Reverting to see if that fixes flakiness of flutter_gallery__transition_perf_with_semantics in devicelab.

Commit:
drop/restore focus when app becomes invisible/visible (#24744)

Commit:
IntrinsicWidth stepWidth or stepHeight == 0.0 (#25228)

Commit:
Revert "drop/restore focus when app becomes invisible/visible" (#25390)

DaveShuckerow committed:
Add ipv6 and observatory port support to the attach command (#25303)

* Add ipv6 and observatory port support to the attach command.

* Remove whitespace

* Explain why a name change is useful here

* Refactor common flags out into the parent

* Add an additional test case for when observatory port is provided but debug port is not

* Remove whitespace

* Fix flag validation

Commit:
Remove TextField.noMaxLength, use maxLength = -1 instead (#24999)

Commit:
Add fuchsia devices to daemon command (#25344)

jslavitz committed:
Adds support for floating cursor (#25384)

* Adds support for floating cursor.

Commit:
Fixed Typography null factory constructor (#24932)

sbaranov committed:
Flutter tool support for automatic saving of JIT compilation trace (#25301)

Commit:
try disabling flutter run test (#25416)

goderbauer committed:
Reland "Call mark* methods before attaching child (#25239)" (#25395)

This reverts commit ea7d086e11e31d32fef1f6217566361f763c1429.

The revert did not fix the flakiness.

engine-flutter-autoroll committed:
Roll engine 101b27da7a1b..494112582932 (19 commits) (#25423)


https://github.com/flutter/engine/compare/101b27da7a1b...494112582932

git log 101b27da7a1b..494112582932 --no-merges --oneline
494112582 Revert &#34;Compile embedder unit test Dart to kernel (#7227)&#34; (flutter/engine#7230)
f7d91d6fa Roll src/third_party/skia 282ec5dc6ca7..2e6db18c4309 (1 commits) (flutter/engine#7228)
ac9e521a1 Compile embedder unit test Dart to kernel (flutter/engine#7227)
828acc739 Roll src/third_party/skia d4962e7e07c0..282ec5dc6ca7 (1 commits) (flutter/engine#7226)
8a7ae957f Undeprecated BigInteger support, but document what it actually does. (flutter/engine#6903)
17789249d Offset.fromDirection and Size.aspectRatio (flutter/engine#6805)
9ae82173a Roll src/third_party/skia 8b78d70b9a10..d4962e7e07c0 (1 commits) (flutter/engine#7225)
a987b1700 Roll src/third_party/skia a1bded9a4f28..8b78d70b9a10 (1 commits) (flutter/engine#7224)
ac0599357 Roll src/third_party/skia 16d00eeef7d1..a1bded9a4f28 (2 commits) (flutter/engine#7223)
91bb1f8f4 Roll src/third_party/skia f391d0f771c6..16d00eeef7d1 (4 commits) (flutter/engine#7222)
6c40f84a5 [vulkan] Fix Fuchsia build
4f148edf0 Roll src/third_party/skia f6206f91b3c1..f391d0f771c6 (4 commits) (flutter/engine#7220)
1bc7ccf60 Support real fonts in &#39;flutter test&#39; (flutter/engine#6913)
5244cf429 Roll src/third_party/skia 59c9f1595ecd..f6206f91b3c1 (1 commits) (flutter/engine#7219)
0f10d3f99 Roll src/third_party/skia 42e7a7ed6511..59c9f1595ecd (3 commits) (flutter/engine#7218)
167642182 Roll src/third_party/skia 21a7be0741ac..42e7a7ed6511 (1 commits) (flutter/engine#7217)
1a2714bbe Adds force cursor support (flutter/engine#6945)
b50e3491c Roll src/third_party/skia cb6b4bde0c2e..21a7be0741ac (3 commits) (flutter/engine#7216)
826f34264 [Fuchsia] Depend on libtrace when that is what&#39;s really meant (flutter/engine#7214)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
Roll engine 494112582932..5af435098d34 (2 commits) (#25438)


https://github.com/flutter/engine/compare/494112582932...5af435098d34

git log 494112582932..5af435098d34 --no-merges --oneline
5af435098 Roll src/third_party/skia cb8157e6266c..e6f635476da3 (1 commits) (flutter/engine#7233)
6c4e90731 Roll src/third_party/skia 2e6db18c4309..cb8157e6266c (1 commits) (flutter/engine#7232)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

DanTup committed:
Support ANDROID_SDK_ROOT in addition to ANDROID_HOME (#25221)

* Fall back to ANDROID_SDK_ROOT if ANDROID_HOME is not set

And update descriptions to use the non-deprecated ANDROID_SDK_ROOT.

Fixes #15114.

* Remove trailing whitespace

* Update dev/devicelab/lib/framework/adb.dart

Co-Authored-By: DanTup <danny@tuppeny.com>

* Reformat long line

Commit:
Update localizations (#25394)

zanderso committed:
[fuchsia] Get Dart VM service ports from The Hub (#25332)

justinmc committed:
Revert "Revert "obscureText and enableInteractiveSelection defaults"" (#25342)

* Revert "Add imports section to sample code templates, and more docs. (#25184)"

This reverts commit 95b01247852e957719c55bd9939dc0c1e1e13e91.

* Revert "Use stderr instead of stdout to contain errors in flutter attach test (#25305)"

This reverts commit 2b819dd257b85179f16ed6f8511c3728c58babec.

* Revert "Allow detection of taps on TabBar (#23919)"

This reverts commit 01694ab62d33d345718d26e32a80db8d553f0ace.

* Revert "a549981da Roll src/third_party/skia a69b10312977..5eb29448dfbd (1 commits) (flutter/engine#7211) (#25333)"

This reverts commit a37099f3b16cf218980f864effa1b99cbd6d2ffd.

* Revert "Revert "obscureText and enableInteractiveSelection defaults (#24527)" (#25335)"

This reverts commit c5457068df5feb8aa0fdecb05de4260962760b2a.

* Password fields are no longer selectable nor copiable

devoncarew committed:
fix the daemon device.getDevices call (#25443)

dnfield committed:
Support Java 1.8 (#25470)

devoncarew committed:
don't warn for non-matching device discoverers (#25440)

* don't warn for non-matching device discoverers

* remove an out of date test

liyuqian committed:
Mark flaky tests as such (#25482)

See https://github.com/flutter/flutter/issues/25481

liyuqian committed:
Add cull opacity perf test to device lab (#25381)

For https://github.com/flutter/flutter/issues/24712

This test verifies that https://github.com/flutter/engine/pull/6923 will speedup the average rasterize time of this test from ~150ms to ~10ms

Please see non-auto-generated files in https://github.com/flutter/flutter/pull/25381/commits/37b21d9fb4756a8d8d12ae0def1d0640740c8818

a14n committed:
fix some formatting issues (#25474)

DanTup committed:
Fix (probably) bad assert when running `flutter test` with asserts and `--start-paused` (#25512)

Fixes #25201.

Commit:
TextField.onChanged() doc update (#25473)

justinmc committed:
Add InputDecoration alignLabelWithHint parameter (#24993)

* InputDecorator param for alignment of label

* Put baseline/center code in the right place where label is layed out

* Fix existing test

* Test top label positioning

* Rename to alignLabelWithHint, and make it a bool

* Test for TextField with and without alignLabelWithHint set

* alignLabelWithHint in theme as well

* debugFillProperties addition and test

* Small style fixes for review

* Fix analyze const error

goderbauer committed:
assert(elevation >= 0.0) and doc clarifications (#25345)

GaryQian committed:
Iterate through potential grapheme cluster lengths in text painter (#24797)

GaryQian committed:
Fix typo (#25237)

sbaranov committed:
Fix gradle local.properties tests that were never excersized. (#25484)

These tests were actually failing, but were silently ignored due overly broad exception catching logic.

Commit:
TransitionRoute.canTransitionFrom,To() doc update (#25477)

a14n committed:
make see also sections uniform (#25513)

sbaranov committed:
Read correct cached VM snapshot in dynamic mode (PRODUCT vs RELEASE) (#25472)

sbaranov committed:
Fix flutter tool to actually honor --build-number/--build-name flags. (#25520)

sbaranov committed:
Update examples to match the new version of generated build.gradle (#25483)

This version honors flutter tool's --build-number/--build-name flags.

engine-flutter-autoroll committed:
Roll engine 5af435098d34..08465e800035 (3 commits) (#25522)


https://github.com/flutter/engine/compare/5af435098d34...08465e800035

git log 5af435098d34..08465e800035 --no-merges --oneline
08465e800 Verify RunConfiguration is valid before running (flutter/engine#6922)
f986a12e2 Correct spelling in doc comments (flutter/engine#7179)
67cd7d4d3 Compile embedder unit test Dart to kernel (flutter/engine#7231)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Commit:
Video demo instrumentation (#25489)

* Video Demo instrumentation

* Video Demo instrumentation

* Updated per review

* Fixed a typo

xster committed:
Adds CupertinoTheme (#23759)

liyuqian committed:
Revert "Update examples to match the new version of generated build.gradle (#25483)" (#25563)

This reverts commit 8acc058cf198cca10a51d31fb743c7ab126a796e.

Reverts #25483

Reason for revert: it seems to break the flutter_gallery_instrumentation_test post-submit test.

TBR: @sbaranov

sbaranov committed:
Fix the build (#25568)

tonyzhao1 committed:
Flutter doctor error message lookup (#23889)

johnsonmh committed:
[Material] Theme-able TextStyles for AlertDialog (#25339)

* Themable elevation on dialogs.

* AlertDialogs Title/Content text styles

* Themable elevation on dialogs.

AlertDialogs Title/Content text styles

* finish merging change

* fixing tests

* Docs fixes

* Change const to final to fix analyzer errors

* Fix analyzer errors

* Remove dart:ui import

* Hans Comments

engine-flutter-autoroll committed:
Roll engine 08465e800035..526e0e3afb2c (10 commits) (#25529)

* 8190e939f Simplify conversion of numeric types in the message codec on iOS (flutter/engine#7097)

* 42c62a944 Fix destruction of the child object list in the iOS accessibility bridge (flutter/engine#7108)

* 951edf3e1 Simplify nested try-with-resources statements (flutter/engine#7239)

* f79f7f631 Allow inferred types using diamond syntax (flutter/engine#7237)

* 5c63303c2 Revert spelling correction in licence matcher (flutter/engine#7242)

* fbce2bf14 Fix linter errors in ResourceUpdater (flutter/engine#7240)

* ba117366e Compute cull_rect and optimize in Layer::Preroll (flutter/engine#6923)

* e8f4968d6 Update iOS unit test for the removal of hex string encoding of uint64 data (flutter/engine#7246)

* 370cc67d9 Roll src/third_party/skia e6f635476da3..4b75254560d4 (61 commits) (flutter/engine#7247)

* 526e0e3af Roll src/third_party/skia 4b75254560d4..67761eb35f97 (6 commits) (flutter/engine#7249)

a14n committed:
fix indentation in doc comments (#25521)

sbaranov committed:
Update examples to match the new version of generated build.gradle (#25483) (#25569)

This version honors flutter tool's --build-number/--build-name flags.

engine-flutter-autoroll committed:
Roll engine 526e0e3afb2c..9edb4f72b914 (9 commits) (#25578)


https://github.com/flutter/engine/compare/526e0e3afb2c...9edb4f72b914

git log 526e0e3afb2c..9edb4f72b914 --no-merges --oneline
9edb4f72b Update dynamic patching terminilogy to match that of the flutter tool (flutter/engine#7248)
2c9fc872d Roll src/third_party/skia 1a237195b527..8d23a584fc43 (8 commits) (flutter/engine#7255)
8b73b9d42 Fix javadoc for Android-28 (flutter/engine#7254)
2e4a02719 Roll src/third_party/skia 192d66440910..1a237195b527 (1 commits) (flutter/engine#7253)
ccaac1af3 Roll src/third_party/skia b933285ebb4e..192d66440910 (3 commits) (flutter/engine#7252)
8939e2abe Fix a misspelling (flutter/engine#7234)
ada6748d0 Extract function to collect licenses for component (flutter/engine#7250)
9f10b10a0 Roll src/third_party/skia 67761eb35f97..b933285ebb4e (1 commits) (flutter/engine#7251)
cfbc94864 Roll buildroot to support Android SDK 28 (flutter/engine#7245)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

sbaranov committed:
Add missing dependency to fix the build (#25582)

xster committed:
Remove code signing special casing for Googlers round 2 (#24580)

GaryQian committed:
Use full textspan tree instead of top level textspan (#25574)

sbaranov committed:
Flutter tool support for building dynamic updates (#25576)

engine-flutter-autoroll committed:
Roll engine 9edb4f72b914..5ec8ef850934 (2 commits) (#25590)


https://github.com/flutter/engine/compare/9edb4f72b914...5ec8ef850934

git log 9edb4f72b914..5ec8ef850934 --no-merges --oneline
5ec8ef850 Remove unused GrContext in AndroidSurfaceGL (flutter/engine#7258)
c08a80ada Roll src/third_party/skia 8d23a584fc43..ce9a901a76eb (10 commits) (flutter/engine#7259)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

jslavitz committed:
Adds force press gesture detector and recognizer (#24554)

* adds Force Press gesture detector and recognizer

Commit:
Update DayPicker,DatePicker doc "see also" sections (#25573)

sbaranov committed:
Don't parse APK unless explicitly requested (#25595)

engine-flutter-autoroll committed:
Roll engine 5ec8ef850934..dbdb5e6f8c9f (7 commits) (#25628)


https://github.com/flutter/engine/compare/5ec8ef850934...dbdb5e6f8c9f

git log 5ec8ef850934..dbdb5e6f8c9f --no-merges --oneline
dbdb5e6f8 Roll src/third_party/skia 428ec1e55c05..44539be30ef8 (1 commits) (flutter/engine#7265)
cf11d86bf Roll src/third_party/skia 21ee517391bc..428ec1e55c05 (3 commits) (flutter/engine#7264)
2d2b9d753 Re-run license tool on all source when it changes (flutter/engine#7260)
5fa1d5980 Roll src/third_party/skia 7757bba636d8..21ee517391bc (1 commits) (flutter/engine#7263)
6a1f01724 Roll src/third_party/skia ce9a901a76eb..7757bba636d8 (4 commits) (flutter/engine#7262)
079d60452 Add a system message channel for controlling the Skia resource cache size (flutter/engine#7257)
41222d055 Update usage of some Android APIs that are deprecated in API level 28 (flutter/engine#7261)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

justinmc committed:
Right aligned backspace bug (#25229)

* Fix bug in calculation of downstream text distance when aligned: right

* After having affinity explained to me, match getOffset and tests to
expected behavior

* Clean up test formatting and fix analyze errors

* Improve test comments and reorganize a bit

* Specify logical pixels

dnfield committed:
Default baseline build options (#25631)

* fix crash

* default baseline

xster committed:
Let CupertinoTabScaffold handle keyboard insets too (#25593)

xster committed:
Fix material reference in CupertinoPicker doc (#25584)

sbaranov committed:
Switch over to the new name for compilation trace native function (#25594)

* Switch over to the new name for compilation trace native function.

Also see: https://github.com/flutter/engine/pull/7256

* rename

* roll the engine

xster committed:
Revise Android and iOS gestures on Material TextField (#24457)

Hixie committed:
[O] Remove many timeouts. (#23531)

* Remove many timeouts.

These are essentially self-inflicted race conditions. Instead of timeouts we're going to try a more verbose logging mechanism that points out when things are taking a long time.

* Get the attach tests to pass.

* Apply review comments from Todd

* More review comment fixes

* Put back the extended timeouts here now that I know why we have them...

Hixie committed:
Revert "[O] Remove many timeouts. (#23531)" (#25646)

This reverts commit 76f70810e457b7a3a39fd59f59a930b5e93c6066.

pranayairan committed:
Adding support for android app bundle - Issue #17829 (#24440)

* adding support for android app bundle.

* removing the debug statement.

* fixing formatting and code review changes.

* Revert "fixing formatting and code review changes."

This reverts commit 2041d459f335242555a0b75e445343134c245494.

* Fixing code formatting issues.

* updating review comments fixing comments and spacing.

* changing and to & to rerun the CI and tests.

* updating the comment to re-run the test

updating the comment to re-run the test

* fixing the formatting.

* updating comments to re-trigger build

updating comments to re-trigger build

engine-flutter-autoroll committed:
Roll engine e859296b71de..6a9041853c4b (4 commits) (#25643)


https://github.com/flutter/engine/compare/e859296b71de...6a9041853c4b

git log e859296b71de..6a9041853c4b --no-merges --oneline
6a9041853 [License] Assert license filename is non-null, non-empty (flutter/engine#7270)
72aa37842 [License] Eliminate duplicate case in switch (flutter/engine#7269)
74af4429a Mark all unreassigned locals final in license tool (flutter/engine#7267)
0310e910d Roll src/third_party/skia 44539be30ef8..b03e024a4034 (7 commits) (flutter/engine#7268)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

engine-flutter-autoroll committed:
Roll engine 6a9041853c4b..54a3577c0139 (23 commits) (#25649)


https://github.com/flutter/engine/compare/6a9041853c4b...54a3577c0139

git log 6a9041853c4b..54a3577c0139 --no-merges --oneline
54a3577c0 Roll src/third_party/skia b03e024a4034..bdfe3a3ee9c3 (14 commits) (flutter/engine#7277)
ead76a4db Remove unused native function dumpCopmpilationTrace(). (flutter/engine#7276)
eaffc462a [License] Enable avoid_positional_boolean_parameters lint (flutter/engine#7275)
b07753bad [License] Sync analysis_options.yaml from framework (flutter/engine#7273)
233c100b6 [License] Enable always_require_non_null_named_parameters
f4b348f77 [License] Enable flutter_style_todos lint
48be433ce [License] Enable prefer_is_empty lint
bd648bad6 [License] Enable prefer_const_constructors lint
2b207a51c [License] Enable unnecessary_parenthesis lint
97ae27164 [License] Enable prefer_initializing_formals lint
c8f83fbea [License] Enable prefer_collection_literals lint
d120e15af [License] Enable prefer_void_to_null lint
e2ab3780a [License] Enable directives_ordering lint
5f2e6cd41 [License] Enable unnecessary_overrides lint
bd955d78a [License] Enable avoid_function_literals_in_foreach_calls lint
392df65f6 [License] Enable prefer_const_declarations lint
06e35912d [License] Enable prefer_final_fields lint
32423ef38 [License] Enable prefer_asserts_in_initializer_lists lint
4703a7f10 [License] Enable prefer_equal_for_default_values lint
1d7285bea [License] Enable empty_catches lint
68fadca0e [License] Enable unnecessary_this lint
c6f5e67cb [License] Enable prefer_single_quotes lint
cc51731d8 Clarify TextAffinity docs (flutter/engine#7238)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

Hixie committed:
Update links for China help (#25238)

We have a page on the Web page now.

a14n committed:
no period after an alone link in see also section (#25604)

amirh committed:
Depend on the goldens repo through git. (#25479)

It's easier to run the android_views integration test this way, without
needing to worry if we have a local clone of the goldens repo or not.

sbaranov committed:
Revert dependency upgrade to see if it helps with build times and APK size (#25642)

sbaranov committed:
Friendlier flags for Dart compilation training. (#25645)

* Renamed --save-compilation-trace to flutter run --train.
* Renamed --precompile=<file> to --compilation-trace-file=<file>.
* In dynamic mode, made JIT snapshot the default, instead of kernel file.

amirh committed:
Pin the goldens repo to a specific commit in the android_views test. (#25678)

Commit:
Updated Shrine demo (#25674)

GaryQian committed:
Expose font fallback API in TextStyle, Roll engine 54a3577c0139..215ca1560088 (8 commits) (#25585)

Hixie committed:
Some minor tweaks to InputDecoration (mainly docs). (#24643)

Hixie committed:
Fix merge conflict. (#25718)

Hixie committed:
Handle errors in `compute()` by propagating them to the Future. (#24848)

Hixie committed:
ClipPath.shape and related fixes (#24816)

Hixie committed:
Undeprecated BigInteger support, but document what it actually does. (#24511)

Also, some code cleanup.

Hixie committed:
Provide some more locations for the FAB. (#24736)

Top left and top right for big FABs, and top left for mini FABs.

engine-flutter-autoroll committed:
Roll engine 215ca1560088..d8c5ec0e11b5 (12 commits) (#25728)


https://github.com/flutter/engine/compare/215ca1560088...d8c5ec0e11b5

git log 215ca1560088..d8c5ec0e11b5 --no-merges --oneline
d8c5ec0e1 Roll src/third_party/skia f16825ed3c71..dc5863c4d665 (1 commits) (flutter/engine#7297)
efb8a3f07 Roll src/third_party/skia 945a56644728..f16825ed3c71 (1 commits) (flutter/engine#7296)
67150a535 Roll src/third_party/skia 28bd882a1e40..945a56644728 (2 commits) (flutter/engine#7295)
3705a1ed3 Roll src/third_party/skia ea8900e74ea7..28bd882a1e40 (1 commits) (flutter/engine#7294)
943ecfc75 Roll src/third_party/skia 13654b20ce5b..ea8900e74ea7 (1 commits) (flutter/engine#7293)
7b12ae241 Roll src/third_party/skia 12a6d452b126..13654b20ce5b (1 commits) (flutter/engine#7292)
1fe7e1fde Roll src/third_party/skia c983480b117f..12a6d452b126 (1 commits) (flutter/engine#7291)
a8b0d263c Roll src/third_party/skia 2e40d9886402..c983480b117f (1 commits) (flutter/engine#7290)
c64f971e3 Roll src/third_party/skia 98a4e4a4135b..2e40d9886402 (1 commits) (flutter/engine#7289)
65420c264 Roll src/third_party/skia 2a0967383087..98a4e4a4135b (1 commits) (flutter/engine#7288)
770536ab6 Only overflow the cache for one required frame (flutter/engine#7048)
b6880bbc9 Roll src/third_party/skia 9803b8f1d197..2a0967383087 (16 commits) (flutter/engine#7285)

The AutoRoll server is located here: https://autoroll.skia.org/r/flutter-engine-flutter-autoroll

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, please contact the current sheriff (liyuqian@google.com), and stop
the roller if necessary.

kangwang1988 commented:
@jason-simmons  Should I resolve the conflicts now?

kangwang1988 committed:
Merge

the4thfloor commented:
@kangwang1988 it looks that something went wrong with your last change(s).

As I remember there was a merge conflict. Have you resolved it with a rebase or with a merge from somewhere else?

The commit count jumped to 162, so I guess you merged master(?) into this commit?

To fix merge conflicts, you should do a [git rebse](https://docs.openstack.org/doc-contrib-guide/additional-git-workflow/rebase.html)

kangwang1988 commented:
@the4thfloor  I did a git rebase and resolve the conflicts.

GregorySech commented:
Currently in stable `--build-number=x --build-name=a.b.c` doesn't override the version specified in pubspec.yaml is this pull request related to it or should I file a new issue?

kangwang1988 commented:
@GregorySech 
You can cherry-pick commit:`df9626c` to see if it solves your issue.

xster commented:
Can you make sure your branch only has 04fb9a27ebea8147cd7937c3a1c3b3a3f22892a5 by rebasing on master instead of merging?

kangwang1988 commented:
This pr is not necessary anymore.
As in commit 693fb6560 by Stanislav, Dec 19, 2018, it's fixed. 
Closing it now.
